### PR TITLE
Project launched in demo mode shows preview mode.

### PR DESCRIPTION
**Pull request by wszczepanski97**
[Problem] When project is in demo mode it should display only preview mode and HTML document. Editor mode is not available.
[Solution] When model of the project is loaded, launch preview mode. Add isDemoVersion variable which is a handler of project type.

Signed-off-by: Wojciech Szczepanski <w.szczepansk@samsung.com>

### CODE REVIEW

File: design-editor/src/pane/design-editor-element.js [hunk 1]
```diff
@@ -294,6 +297,7 @@ class DesignEditor extends DressElement {
 			}
 
 			this._$iframe.one('load', () => {
+				const isDemoVersion = PreferencesManager.getViewState('projectType') === 'demo' ? true : false;
```
Comment (hsiwkin) [inline design-editor/src/pane/design-editor-element.js]: You can just write: 
`const isDemoVersion = PreferencesManager.getViewState('projectType') === 'demo'`
means the same, but it's shorter

File: design-editor/src/system/stage-manager.js [hunk 2]
```diff
@@ -237,7 +239,8 @@ class StageManager {
      */
     _togglePreview(toggleEditor) {
         var $workSpace = $(editor.selectors.workspace),
-            preview = null;
+			preview = null,
+			isDemoVersion = PreferencesManager.getViewState('projectType') === 'demo' ? true : false;
```
Comment (hsiwkin) [inline design-editor/src/system/stage-manager.js]: to stay DRY I'd create a util file in utils catalog and add there a function isDemoVersion which would be used both here and in design-editor-element.

File: design-editor/src/pane/design-editor-element.js [hunk 3]
```diff
@@ -30,10 +31,12 @@ import path, {relative, join} from 'path';
...
 	DELETE: 46
-};
-const INTERNAL_ID_ATTRIBUTE = 'data-id';
-const LOCK_CLASS = 'lock';
-const RE_COPY_ATTRIBUTE_TO_CONTAINER = new RegExp('^(data-(?!tau)|st-).*');
+	},
+	INTERNAL_ID_ATTRIBUTE = 'data-id',
+	LOCK_CLASS = 'lock',
+	RE_COPY_ATTRIBUTE_TO_CONTAINER = new RegExp('^(data-(?!tau)|st-).*'),
+	brackets = window.brackets || window.top.brackets,
```
Comment (korneliakobiela) [inline design-editor/src/pane/design-editor-element.js]: Please use our function `checkGlobalContext()` from utils module to choose if some package get from window or window top.
Example
``` javascript
brackets = utils.checkGlobalContext('brackets');
```

File: design-editor/src/system/stage-manager.js [hunk 4]
```diff
@@ -17,9 +17,10 @@ import {PreviewToolbarElement} from '../panel/preview/preview-toolbar-element';
 // import {InteractionViewToolbarElement} from '../panel/preview/interaction-view-toolbar-element';
 import {InfoElement} from '../pane/select-layer/info-element';
 
-const CompositeDisposable = editor.CompositeDisposable;
-const DESIGN = ViewType.Design;
-const vscode = window.vscode;
+const CompositeDisposable = editor.CompositeDisposable,
+	DESIGN = ViewType.Design,
+	brackets = window.brackets || window.top.brackets,
+	PreferencesManager = brackets ? brackets.getModule('preferences/PreferencesManager') : {};
```
Comment (korneliakobiela) [inline design-editor/src/system/stage-manager.js]: I think we don't need these conditions on brackets.getModule().
AFAIK we mocked brackets API and brackets will always return an object.

File: design-editor/src/system/stage-manager.js [hunk 5]
```diff
@@ -260,11 +262,18 @@ class StageManager {
 
             if (window.atom !== undefined) {
```
Comment (korneliakobiela) [inline design-editor/src/system/stage-manager.js]: We don't support atom anymore. You can refactor this if .. else statement and stay DRY rule

**Pull request merged**